WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java

```diff
@@ -2321,30 +2321,45 @@ private void showLoadingProgress(boolean showProgress) {
         }
     }
 
-    private void setIsUpdating(boolean isUpdating, UpdateAction updateAction) {
-        if (!isAdded() || mIsUpdating == isUpdating) {
-            if (!isUpdating) {
-                mRecyclerView.setRefreshing(false);
-            }
-            return;
-        }
+    private void clearCurrentUpdateActions() {
+        if (!isAdded() || !isUpdating()) return;
 
-        if (updateAction == UpdateAction.REQUEST_OLDER) {
-            // show/hide progress bar at bottom if these are older posts
-            showLoadingProgress(isUpdating);
-        } else if (isUpdating) {
-            // show swipe-to-refresh if update started
-            mRecyclerView.setRefreshing(true);
+        mCurrentUpdateActions.clear();
+        updateProgressIndicators();
+    }
+
+    private void setIsUpdating(boolean isUpdating, @NonNull UpdateAction updateAction) {
+        if (!isAdded()) return;
+
+        boolean isUiUpdateNeeded;
```

Comment:
Wondering if we should synchronise this block of code on mCurrentUpdateActions.

Reply:
Guess it's not an issue since we're already interacting with the UI on the main thread.

Reply:
Yeah, I think currently we are doing most stuff in this Fragment in the main thread (even stuff that shouldn't) but this is something to think about when we decide to refactor the Reader, even though we will likely use a more recent tech stack with LiveData/Flows and Coroutines in the ViewModel to propagate this kind of UI state.